test/index.test.ts

```diff
@@ -771,6 +771,211 @@ describe('Unit | Utility | changeset', () => {
+        expect(changeset.changes).toEqual([{ key: 'contact.emails.0', value: 'fred@email.com' }]);
+        expect(changeset.get('contact.emails').unwrap()).toEqual(['fred@email.com']);
+
+        changeset.rollback();
+
+        expect(changeset.get('contact.emails.0')).toEqual('bob@email.com');
+        expect(changeset.changes).toEqual([]);
+        expect(changeset.get('contact.emails')).toEqual(['bob@email.com']);
+      });
+
+      it('can be saved', () => {
+        const changeset = Changeset(initialData);
+
+        changeset.set('contact.emails.0', 'fred@email.com');
+
+        expect(changeset.get('contact.emails.0')).toEqual('fred@email.com');
+        expect(changeset.changes).toEqual([{ key: 'contact.emails.0', value: 'fred@email.com' }]);
+
+        changeset.save();
+
+        expect(changeset.get('contact.emails.0')).toEqual('fred@email.com');
+        expect(changeset.changes).toEqual([]);
+      });
+
+      it('can add items to the array', () => {
+        const changeset = Changeset(initialData);
+
+        changeset.set('contact.emails.1', 'fred@email.com');
+
+        expect(changeset.get('contact.emails.1')).toEqual('fred@email.com');
+        expect(changeset.get('contact.emails').unwrap()).toEqual([
+          'bob@email.com',
+          'fred@email.com'
+        ]);
+        expect(changeset.changes).toEqual([{ key: 'contact.emails.1', value: 'fred@email.com' }]);
+
+        changeset.set('contact.emails.3', 'greg@email.com');
+
+        expect(changeset.get('contact.emails.3')).toEqual('greg@email.com');
+        expect(changeset.get('contact.emails').unwrap()).toEqual([
+          'bob@email.com',
+          'fred@email.com',
+          undefined,
+          'greg@email.com'
+        ]);
+        expect(changeset.changes).toEqual([
+          { key: 'contact.emails.1', value: 'fred@email.com' },
+          { key: 'contact.emails.3', value: 'greg@email.com' }
+        ]);
+      });
+
+      it('can remove items from the array', () => {
+        const changeset = Changeset(initialData);
+
+        changeset.set('contact.emails.1', 'fred@email.com');
+
+        expect(changeset.get('contact.emails.1')).toEqual('fred@email.com');
+        expect(changeset.get('contact.emails').unwrap()).toEqual([
+          'bob@email.com',
+          'fred@email.com'
+        ]);
+        expect(changeset.changes).toEqual([{ key: 'contact.emails.1', value: 'fred@email.com' }]);
+
+        changeset.set('contact.emails.0', null);
+
+        expect(changeset.get('contact.emails.0')).toEqual(null);
+        expect(changeset.get('contact.emails').unwrap()).toEqual([null, 'fred@email.com']);
+        expect(changeset.changes).toEqual([
+          { key: 'contact.emails.0', value: null },
+          { key: 'contact.emails.1', value: 'fred@email.com' }
+        ]);
+
+        changeset.set('contact.emails.1', null);
+
+        expect(changeset.get('contact.emails').unwrap()).toEqual([null, null]);
```

Comment:
idk if this is technically "deleting", but it made me wonder if changeset needs some post-processing callback to run before "save".

cause for some uses, you may want to collapse all the falsey values in an array, like, 
```ts
data.array = data.array.filter(Boolean);
```
but I could also see some use cases where you'd want the falsey values

Reply:
I think in this case, the recommended way would be to `.set('contact.emails', [])`?  I'm happy leaving the nulls in there if that is what the `set` did!